### PR TITLE
metrics: Expose server status

### PR DESCRIPTION
**Pull request by pierreprinetti**
New metric openstack_server: OpenStack servers by status. Useful to track ERRORed instances.

### CODE REVIEW

File: metrics.sh [hunk 1]
```diff
@@ -25,3 +25,13 @@ for service in 'compute' 'network' 'volume'; do
 		echo "${metric}{cloud=\"${OS_CLOUD}\",project=\"${os_project}\",type=\"limit\"} $limit"
 	done
 done
+
+
+metric=openstack_server
```
Comment (mdbooth) [inline metrics.sh:30]: Should probably be `openstack_server_status`

File: metrics.sh [hunk 2]
```diff
@@ -25,3 +25,13 @@ for service in 'compute' 'network' 'volume'; do
...
+echo "# HELP ${metric} OpenStack servers by status"
+echo "# TYPE ${metric} gauge"
+openstack server list -f value -c Status \
+	| uniq -c \
+	| while read -r number state; do
+		echo "${metric}{cloud=\"${OS_CLOUD}\",project=\"${os_project}\",type=\"${state}\"} $number";
```
Comment (mdbooth) [inline metrics.sh:36]: `type` here should probably be `status`.

**Review by mdbooth — CHANGES_REQUESTED** [review]

File: metrics.sh (hunk 1 above)
Comment (pierreprinetti) [reply metrics.sh:30]: It was in my first iteration. But then I thought: shall we reflect the attributes in the metric? That's really the number of existing servers. Shall the metric name reflect the fact that the list is broken down by status?

File: metrics.sh (hunk 2 above)
Comment (pierreprinetti) [reply metrics.sh:36]: yes

File: metrics.sh (hunk 1 above)
Comment (mdbooth) [reply metrics.sh:30]: Ok, I'm sold.

/lgtm

**Pull request merged**